caproto/threading/client.py

```diff
@@ -487,7 +736,7 @@ def received(self, bytes_recv, address):
         for c in commands:
             self._process_command(c)
 
-        return num_bytes_needed
+        return len(bytes_recv) > 0
```

Comment:
This seems like a pretty big change.  Dan and I had long conversations about how this gave us a path to write smarter clients in the future that could be better about memory management.

Reply:
OK, this can be changed to an exception or an error code, I suppose.

Reply:
@klauer Or this could return `(num_bytes_needs, bytes_received)`? If the only reason to want `bytes_received` is to notice when it's `0` then I think I prefer an Exception, but I thought I'd flag the possibility of doing it this way.